pkg/filesystem/local_directory.go

```diff
@@ -51,7 +51,7 @@ func (d *localDirectory) enter(name string) (*localDirectory, error) {
 	}
 	defer runtime.KeepAlive(d)
 
-	fd, err := unix.Openat(d.fd, name, unix.O_DIRECTORY|unix.O_NOFOLLOW|unix.O_RDONLY, 0)
+	fd, err := unix.Openat(d.fd, name, unix.O_DIRECTORY|unix.O_NOFOLLOW|unix.O_PATH|unix.O_RDONLY, 0)
```

Comment:
`O_PATH` is specific to Linux, so we shouldn't use that here.

What are your thoughts on extending the if-statement below to also contain something like this?
```go
if runtime.GOOS == "linux" && err == syscall.ELOOP {
        // Linux 3.10 returns ELOOP, while Linux 4.15 returns ENOTDIR. Prefer the latter.
        return nil, syscall.ENOTDIR
}
```

Reply:
> `O_PATH` is specific to Linux, so we shouldn't use that here.
> 
> What are your thoughts on extending the if-statement below to also contain something like this?
> 
> ```go
> if runtime.GOOS == "linux" && err == syscall.ELOOP {
>         // Linux 3.10 returns ELOOP, while Linux 4.15 returns ENOTDIR. Prefer the latter.
>         return nil, syscall.ENOTDIR
> }
> ```

That sounds like a suitable solution to this issue, we'll update the PR accordingly!